src/Microsoft.IdentityModel.JsonWebTokens/Json/JsonWebToken.PayloadClaimSet.cs

```diff
@@ -112,5 +48,74 @@ internal JsonClaimSet CreatePayloadClaimSet(ReadOnlySpan<byte> byteSpan)
 
             return new JsonClaimSet(claims);
         }
+
+        private protected virtual void ReadPropertyValue(ref Utf8JsonReader reader, Dictionary<string, object> claims)
```

Comment:
We will need to make sure the TryGetPayload still works for custom claims.
For example, the derived token handles the foo claim, TryGetPayload<string>("foo") should have a path to work.

Reply:
For now let's make the method internal protected.